runtime/v2/manager.go

```diff
@@ -312,41 +312,30 @@ func (m *ShimManager) resolveRuntimePath(runtime string) (string, error) {
 		return path.(string), nil
 	}
 
-	var (
-		cmdPath string
-		lerr    error
-	)
-
-	binaryPath := shimbinary.BinaryPath(runtime)
-	if _, serr := os.Stat(binaryPath); serr == nil {
-		cmdPath = binaryPath
-	}
+	cmdPath, err := exec.LookPath(name)
+	if err != nil {
+		if eerr, ok := err.(*exec.Error); ok {
+			if eerr.Err == exec.ErrNotFound {
+				self, err := os.Executable()
+				if err != nil {
+					return "", err
+				}
 
-	if cmdPath == "" {
-		if cmdPath, lerr = exec.LookPath(name); lerr != nil {
-			if eerr, ok := lerr.(*exec.Error); ok {
-				if eerr.Err == exec.ErrNotFound {
-					self, err := os.Executable()
-					if err != nil {
-						return "", err
-					}
-
-					// Match the calling binaries (containerd) path and see
-					// if they are side by side. If so, execute the shim
-					// found there.
-					testPath := filepath.Join(filepath.Dir(self), name)
-					if _, serr := os.Stat(testPath); serr == nil {
-						cmdPath = testPath
-					}
-					if cmdPath == "" {
-						return "", fmt.Errorf("runtime %q binary not installed %q: %w", runtime, name, os.ErrNotExist)
-					}
+				// Match the calling binaries (containerd) path and see
+				// if they are side by side. If so, execute the shim
+				// found there.
+				testPath := filepath.Join(filepath.Dir(self), name)
+				if _, serr := os.Stat(testPath); serr == nil {
+					cmdPath = testPath
+				}
+				if cmdPath == "" {
+					return "", fmt.Errorf("runtime %q binary not installed %q: %w", runtime, name, os.ErrNotExist)
 				}
 			}
 		}
 	}
 
-	cmdPath, err := filepath.Abs(cmdPath)
+	cmdPath, err = filepath.Abs(cmdPath)
```

Comment:
Once the code reaches here, `cmdPath` will be the absolute path right? So is this step necessary.

Reply:
https://pkg.go.dev/os/exec#LookPath
> In older versions of Go, LookPath could return a path relative to the current directory. As of Go 1.19, LookPath will instead return that path along with an error satisfying errors.Is(err, ErrDot). 

The minimum version required for containerd is 1.18
https://github.com/containerd/containerd/blob/34d078e99fbdb8c28feec359634335a2c684e703/go.mod#L1-L3